### PR TITLE
Prj 773 switch to kotest. Small jvm modules tests refactoring

### PR DESCRIPTION
**Pull request by Avvessalom**
Rewriting tests in small jvm modules.

- projector-ij-common
- projector-util-loading

### CODE REVIEW

File: projector-util-loading/src/test/kotlin/org/jetbrains/projector/util/loading/ReflectionTest.kt [hunk 1]
```diff
@@ -23,41 +23,40 @@
...
     @Suppress("unused")
     private var unprotectMeToo = 0
   }
 
-  @Test
-  fun testFieldUnprotectFinal() {
-    runFieldUnprotectTest("unprotectMe", true)
-  }
+  init {
+    test("field should be able to become unprotected") {
```
Comment (ARTI1208) [inline projector-util-loading/src/test/kotlin/org/jetbrains/projector/util/loading/ReflectionTest.kt:41]: Better to name something like "final field can be unprotected". There are additional steps for final fields, so better specify `final` here then `non-final` in next test

File: projector-util-loading/src/test/kotlin/org/jetbrains/projector/util/loading/ReflectionTest.kt [hunk 2]
```diff
@@ -23,41 +23,40 @@
...
+      runFieldUnprotectTest("unprotectMe", true)
+    }
 
-  @Test
-  fun testFieldUnprotectNonFinal() {
-    runFieldUnprotectTest("unprotectMeToo", false)
+    test("non-final field should be able to become unprotect") {
```
Comment (ARTI1208) [inline projector-util-loading/src/test/kotlin/org/jetbrains/projector/util/loading/ReflectionTest.kt:45]: Also better to name something like "field can be unprotected"

**Review by ARTI1208 — CHANGES_REQUESTED** [review]

**Review by SerVB — CHANGES_REQUESTED** [review]
For JVM, could we use annotation spec to minimize the diff? Or do you see any advantages of fun spec?

**Comment by Avvessalom** [comment]
Certainly, we can use Annotations spec. The Fun spec is chosen here for test uniformity, since Annotations spec does not work in multiplatform modules.

**Comment by Avvessalom** [comment]
I rewrote everything on the Annotations spec. I think now the diff is as readable as possible.

File: projector-util-loading/build.gradle.kts [hunk 3]
```diff
@@ -29,6 +29,8 @@ plugins {
 
 setupJacoco()
 
+val kotestVersion: String by project
```
Comment (ARTI1208) [inline projector-util-loading/build.gradle.kts:32]: Move this version near the other ones pls

**Review by ARTI1208 — CHANGES_REQUESTED** [review]
Looks nice

I see that IDEA marks test functions as unused, guess we should report this
изображение

File: projector-ij-common/src/test/kotlin/org/jetbrains/projector/agent/init/IjArgsTest.kt [hunk 4]
```diff
@@ -47,21 +48,21 @@ class IjArgsTest {
...
 
     val map = argString.toArgsMap()
 
-    assertEquals(5, map.size)
-    assertEquals("stringValue", map["key1"])
-    assertEquals("2", map["key2"])
-    assertEquals("3.14", map["key3"])
-    assertEquals("false", map["key4"])
-    assertEquals("customObject", map["key5"])
+    map.size shouldBe 5
```
Comment (SerVB) [inline projector-ij-common/src/test/kotlin/org/jetbrains/projector/agent/init/IjArgsTest.kt:61]:
```suggestion
    map shouldHaveSize 5
```

**Review by SerVB — APPROVED** [review]

**Review by ARTI1208 — APPROVED** [review]

**Pull request merged**